### ISSUE TITLE
Error when toggling `disabled` prop on `Button` component using state

### ISSUE THREAD

**Issue by anthonyn60** [tamagui/tamagui#2028]
### Current Behavior

If the `disabled` prop of a `Button` is tied to a state, toggling the value from `true` to `false` causes the following error:
```
 ERROR  TypeError: Cannot read property 'length' of undefined

This error is located at:
    in Button
    in Unknown (created by Themed(Anonymous))
    in Theme (created by Themed(Anonymous))
    in Themed(Anonymous) (created by Home)
    in RCTView (created by YStack)
    in YStack (created by Home)
..............(the rest of the stack trace)
    in main(RootComponent), js engine: hermes
```

This error is preceeded by the following:

```
Warning: React has detected a change in the order of Hooks called by Button. This will lead to bugs and errors if not fixed. For more information, read the Rules of Hooks: https://reactjs.org/link/rules-of-hooks

   Previous render            Next render
   ------------------------------------------------------
1. useId                      useId
2. useContext                 useContext
3. useRef                     useRef
4. useRef                     useRef
5. useState                   useState
6. useRef                     useRef
7. useContext                 useContext
8. useMemo                    useMemo
9. useRef                     useRef
10. useSyncExternalStore      useSyncExternalStore
11. useCallback               useCallback
12. useEffect                 useEffect
13. useMemo                   useRef
   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

### Expected Behavior

The button should go from disabled to enabled when the `disabled` value is toggled from `true` to `false`.

### Tamagui Version

```markdown
1.82.0
```

### Platform (Web, iOS, Android)

```markdown
iOS
```

### Reproduction

I set up an Expo+Tamagui repo using https://github.com/ivopr/tamagui-expo.

I then added a state called `disabled` (with an updater) to the `Home` component (`app/index.tsx`) with a default value of `true`.

I then updated the first button ("Go to user page") to have its `disabled` value set to the state variable.

I then updated the second button's `onPress` to set the `disabled` state to `false`.

Ran the app with Expo on the emulator and got the error (`yarn run ios`).

Source for `app/index.tsx`:

```markdown
import { useState } from "react";
import { Github, Twitter } from "@tamagui/lucide-icons";
import { Link, useRouter } from "expo-router";
import {
  Button,
  H1,
  ListItem,
  Paragraph,
  Separator,
  YGroup,
  YStack
} from "tamagui";

import { MySafeAreaView } from "../components/MySafeAreaView";
import { MyStack } from "../components/MyStack";

export default function Home() {
  const [disabled, setDisabled] = useState(true); // eslint-disable-line @typescript-eslint/no-unused-vars
  const router = useRouter();

  return (
    <MySafeAreaView>
      <MyStack>
        <YStack
          space="$4"
          maxWidth={600}
        >
          <H1 textAlign="center">Welcome to Tamagui.</H1>
          <Paragraph textAlign="center">
            Here&apos;s a basic starter to show navigating from one screen to
            another.
          </Paragraph>
        </YStack>

        <YStack space="$2.5">
          <Button
            onPress={() => router.push("/users/testuser")}
            disabled={disabled}
          >
            Go to user page
          </Button>
          <Button onPress={() => setDisabled(false)}>Go to tabs page</Button>
        </YStack>

(the rest of the file is the same)
```
```


### System Info

```markdown
System:
    OS: macOS 14.1.1
    CPU: (8) x64 Apple M1 Pro
    Memory: 322.56 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 16.20.1 - ~/.nvm/versions/node/v16.20.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.20.1/bin/yarn
    npm: 8.19.4 - ~/Documents/github/test/yourappname/node_modules/.bin/npm
    bun: 1.0.8 - ~/.bun/bin/bun
  Browsers:
    Chrome: 120.0.6099.129
    Safari: 17.1
  npmPackages:
    @babel/core: ^7.19.3 => 7.23.7
    @google/semantic-release-replace-plugin: 1.1.0 => 1.1.0
    @manypkg/cli: ^0.20.0 => 0.20.0
    @semantic-release/changelog: 6.0.1 => 6.0.1
    @semantic-release/git: 10.0.1 => 10.0.1
    @semantic-release/github: 8.0.6 => 8.0.6
    @semantic-release/npm: 9.0.1 => 9.0.1
    @semantic-release/release-notes-generator: 10.0.3 => 10.0.3
    @tamagui/animations-react-native: ^1.81.4 => 1.82.0
    @tamagui/babel-plugin: ^1.81.4 => 1.82.0
    @tamagui/config: ^1.81.4 => 1.82.0
    @tamagui/font-inter: ^1.81.4 => 1.82.0
    @tamagui/lucide-icons: ^1.81.4 => 1.82.0
    @tamagui/themes: ^1.81.4 => 1.82.0
    @types/react: ~18.2.14 => 18.2.46
    @typescript-eslint/eslint-plugin: ^5.49.0 => 5.62.0
    @typescript-eslint/parser: ^5.49.0 => 5.62.0
    babel-plugin-transform-inline-environment-variables: ^0.4.4 => 0.4.4
    conventional-changelog-conventionalcommits: 5.0.0 => 5.0.0
    eslint: ^8.21.0 => 8.56.0
    eslint-config-prettier: ^8.6.0 => 8.10.0
    eslint-plugin-prettier: ^4.2.1 => 4.2.1
    eslint-plugin-react: ^7.32.1 => 7.33.2
    eslint-plugin-simple-import-sort: ^10.0.0 => 10.0.0
    expo: 49.0.21 => 49.0.21
    expo-constants: ~14.4.2 => 14.4.2
    expo-font: 11.4.0 => 11.4.0
    expo-linear-gradient: ~12.3.0 => 12.3.0
    expo-linking: ~5.0.2 => 5.0.2
    expo-router: 2.0.14 => 2.0.14
    expo-splash-screen: ~0.20.5 => 0.20.5
    expo-status-bar: ~1.6.0 => 1.6.0
    prettier: ^2.8.3 => 2.8.8
    react: 18.2.0 => 18.2.0
    react-dom: 18.2.0 => 18.2.0
    react-native: 0.72.6 => 0.72.6
    react-native-gesture-handler: ~2.12.0 => 2.12.1
    react-native-reanimated: ~3.3.0 => 3.3.0
    react-native-safe-area-context: 4.6.3 => 4.6.3
    react-native-screens: ~3.22.0 => 3.22.1
    react-native-svg: 13.9.0 => 13.9.0
    semantic-release: 19.0.5 => 19.0.5
    tamagui: ^1.81.4 => 1.82.0
    typescript: ^5.3.3 => 5.3.3
```
```

**Comment by natew** [comment]
Ah good catch, fix going out here 742786161d696419beaae3e4204553aadd9ae564 in just a minute

**Comment by anthonyn60** [comment]
Appreciate the quick fix @natew !
